MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioFactory.java

```diff
@@ -104,166 +99,212 @@ public static AtBScenario createScenario(Campaign c, Lance lance, int type, bool
 	}
 
 	@SuppressWarnings("unchecked")
-	public static boolean registerScenario(IAtBScenario scenario) {
+	public static void registerScenario(IAtBScenario scenario) {
 	    final String METHOD_NAME = "registerScenario(IAtBScenario)"; //$NON-NLS-1$
 
-		if (!IAtBScenario.class.isAssignableFrom(scenario.getClass())) {
-	        MekHQ.getLogger().log(AtBScenarioFactory.class, METHOD_NAME, LogLevel.ERROR,
-	                String.format("Unable to register an AtBScenario of class '%s' because is does not implement '%s'.", //$NON-NLS-1$
-	                        scenario.getClass().getName(), IAtBScenario.class.getName()));
-			return false;
-		}
-
 		if (!scenario.getClass().isAnnotationPresent(AtBScenarioEnabled.class)) {
             MekHQ.getLogger().log(AtBScenarioFactory.class, METHOD_NAME, LogLevel.ERROR,
                     String.format("Unable to register an AtBScenario of class '%s' because is does not have the '%s' annotation.", //$NON-NLS-1$
                             scenario.getClass().getName(), AtBScenarioEnabled.class.getName()));
-			return false;
-		}
-
-		int type = scenario.getScenarioType();
-        List<Class<IAtBScenario>> list = scenarioMap.computeIfAbsent(type, k -> new ArrayList<>());
-
-        list.add((Class<IAtBScenario>) scenario.getClass());
+		} else {
+            int type = scenario.getScenarioType();
```

Comment:
This whitespace looks odd.

Reply:
That looks like a tab mixed with spaces, I'll fix it once I can switch branches.

Reply:
The whole files was tab spaced originally, fixed it but it is going to be hard to review if you don't ignore that commit